commands/output/output.go

```diff
@@ -108,13 +109,27 @@ func showThread(r *review.Review, thread review.CommentThread) error {
 		}
 		lines := strings.Split(contents, "\n")
 		if comment.Location.Range.StartLine <= uint32(len(lines)) {
```

Comment:
Optional feedback:

This check should probably also validate the end line before we try to display it; otherwise the tool would die when displaying a thread if any of the comments have invalid locations. The simplest approach would probably be to make `checkCommentLocation` into a method on the "Location" struct, so that here we can call something like `if err := comment.Location.Check(r.Repo); err != nil {`. We would then also do something similar in `commands/comment.go`

This is all optional as far as this change is concerned. If you want to throw that in, then feel free; otherwise I'll put something together as a cleanup later on.

Reply:
I'll look at that tonight, definitely don't want to leave any potential out of bounds accesses hanging around.

Reply:
I have pushed an attempt at this. It seems to do the job. Feels a little odd checking the Repo against the Location, rather than the other way around, but it ends up quite tidy. I'm sure there's a design principle in there I should already know :)

Reply:
I like it; your code is very clean and nicely encapsulates the check. Thanks for adding that in!

As for the order of checking seeming odd; I can definitely see your argument. The only design principle I can think of for this order is to keep the `repository.Repo` interface as small as possible so that it is easy to define alternative implementations.

For instance, in the future we may want to switch from calling the `git` command to using an implementation of git written in Go. By keeping the Repo interface limited to the bare minimum we make such a future migration easier.